### PR TITLE
Ensure UTF8mb4 conversion command always enables tracking again

### LINKED ISSUE #18441: When convert to utf8mb4 command fails, enable tracking again
When running the `core:convert-to-utf8mb4` command, this command may fail eg because of a reason like below:

> Error: {"message":"SQLSTATE[42000]: Syntax error or access violation: 1118 Row size too large. The maximum row size for the used table type, not counting BLOBs, is 65535. This includes storage overhead, check the manual. You have to change some columns to TEXT or BLOBs","file":"\/var\/www\/html\/libs\/Zend\/Db\/Adapter\/Pdo\/Abstract.php","line":280,"backtrace":" on \/var\/www\/html\/libs\/Zend\/Db\/Adapter\/Pdo\/Abstract.php(280)\n#0 \/var\/www\/html\/plugins\/CoreUpdater\/Commands\/ConvertToUtf8mb4.php(91): 

Afterwards, the tracking is disabled because the command sets `record_statistics=0` and only enables it if the command succeeds. If any query fails, we would also expect the setting to be enabled again.

It might otherwise take a user a few days that tracking may be disabled and then data is lost.

In some case we just got alerted instantly that tracking isn't working anymore and we thought there may be a DB issue when in reality tracking was simply disabled.

### PR DESCRIPTION
**Pull request by sgiehl**
### Description:

fixes #18441 

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php [hunk 1]
```diff
@@ -86,23 +86,25 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
             $output->writeln("\n" . Piwik::translate('CoreUpdater_ConsoleStartingDbUpgrade'));
 
-            foreach ($queries as $query) {
-                $output->write("\n" . 'Executing ' . $query . '... ');
-                Db::get()->exec($query);
-                $output->write(' done.');
+            try {
+                throw new \Exception();
```
Comment (justinvelluppillai) [inline plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php]: Testing line?

File: plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php [hunk 2]
```diff
@@ -86,23 +86,25 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+                }
+
+                $output->writeln("\n" . 'Updating used database charset in config.ini.php.');
+                $config                      = Config::getInstance();
+                $config->database['charset'] = 'utf8mb4';
+            } finally {
+                if (!$keepTracking) {
+                    $output->writeln("\n" . Piwik::translate('Enabling Matomo Tracking'));
+                    $config->Tracker['record_statistics'] = '1';
+                    $config->forceSave();
```
Comment (justinvelluppillai) [inline plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php]: `forceSave()` should be outside the if statement to save the charset also?

**Review by justinvelluppillai — CHANGES_REQUESTED** [review]
Made two comments needing fixing, seems fine otherwise.

File: plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php (hunk 1 above)
Comment (sgiehl) [reply plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php]: Ooops :grimacing:

File: plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php (hunk 2 above)
Comment (sgiehl) [reply plugins/CoreUpdater/Commands/ConvertToUtf8mb4.php]: ah, you're right. changed that.

**Review by justinvelluppillai — APPROVED** [review]

**Comment by sgiehl** [comment]
@justinvelluppillai I have pushed another small fix. In theory it could have happened that `$config` might not have been defined in the `finally`, if `$keepTracking` is `false`, and one of the db queries fails.

**Pull request merged**